Update ucsc-fatovcf from v407 to v426; remove skip.

AngieHinrichs opened the pull request:
@BiocondaBot please add label

ucsc-fatovcf has not been updated since March 2021, and it has `skip: True # [osx]`.  However, the tool pangolin, used by public health departments worldwide to assign SARS-CoV-2 lineages to new genome sequences, depends on faToVcf when using --usher mode, which is slated to become pangolin's default mode soon.  In June 2021, an important new option was added to faToVcf; pangolin really should be using the latest version of faToVcf with the new option.

This PR updates the version from 407 to the latest stable version, 426, and removes the `skip: True # [osx]`.

I expect there will be some build problems because the `skip` must have been added for a reason, but I'm sure we can work through them.

dpryan79 commented:
The issue I'm running into with all of the UCSC packages right now is that they don't compile due to having duplicate symbols. I suspect we need to patch all of them. That's what happened with OSX, since clang has complained about that for a long time. It seems that GCC 10 also now throws an error due to it.

dpryan79 commented:
The issue I'm running into with all of the UCSC packages right now is that they don't compile due to having duplicate symbols. I suspect we need to patch all of them. That's what happened with OSX, since clang has complained about that for a long time. It seems that GCC 10 also now throws an error due to it.

AngieHinrichs commented:
Thanks @dpryan79 -- I'm keen to fix it but I'm not familiar enough with the conda build process to find the specific error messages about duplicate symbols, or exactly how the test is failing.  I can see the log file https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=12445&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572 -- are there lines of that file that refer to the duplicate symbols?  If not, is there a way to inspect a file that has those error messages?  Should I try to build & test locally on my (Mac OSX) laptop?

dpryan79 commented:
The issue I'm running into with all of the UCSC packages right now is that they don't compile due to having duplicate symbols. I suspect we need to patch all of them. That's what happened with OSX, since clang has complained about that for a long time. It seems that GCC 10 also now throws an error due to it.

dpryan79 commented:
Hmm, seems the issue is just with the test with this one. Not sure why it's failing.

dpryan79 commented:
You're certainly welcome to try building and testing locally. You can `conda create -n bioconda bioconda-utils` and then use `bioconda-utils build --packages ucsc-fatovcf` to test.

dpryan79 commented:
You're certainly welcome to try building and testing locally. You can `conda create -n bioconda bioconda-utils` and then use `bioconda-utils build --packages ucsc-fatovcf` to test.

AngieHinrichs commented:
The compile was definitely failing before I added "USE_HIC=0" to disable an unneeded extra feature ([fedaa40](https://github.com/bioconda/bioconda-recipes/pull/33369/commits/fedaa40d7092921504b72e95d3d2d9bf5b759146)), in case that helps with other UCSC packages.

> Hmm, seems the issue is just with the test with this one. Not sure why it's failing.

Is there a way to see specific test commands that are executed and the error messages?  https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=12445&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=969 just says "Tests failed"

The meta.yaml test is just "which {{ program }}":
```
test:
  commands:
    # just check for existence, because the individual packages have no unified behavior
    - which {{ program }}
```

There is also a run_test.sh script that runs the commands with no options, which causes the usage command to be printed and faToVcf exits -1, which the script seems to expect:
```
#!/bin/bash
faToVcf 2> /dev/null || [[ "$?" == 255 ]]
```

But I don't see anything in the log file that indicates whether either of those things are executed and what the failure is.

dpryan79 commented:
You could add `set -x` to the `run_test.sh`, then you'll see the command it's using.

dpryan79 commented:
You could add `set -x` to the `run_test.sh`, then you'll see the command it's using.

AngieHinrichs commented:
OK, thanks for the suggestion!  I added `set -x` to run_test.sh and also added just `faToVcf` with no output redirection.  Now the usage statement from faToVcf appears in the log: https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=12490&view=logs&j=e14e69ff-a0ae-55c4-b71d-229b239cfb2f&t=7df82132-b284-504b-53d6-7d3e63519572&l=1087
It seems to stop there (faToVcf returns -1 with no args) so if the script is running with `-e` that makes sense.  I will remove the plain old `faToVcf`.  The next iteration should show `faToVcf 2> /dev/null || [[ "$?" == 255 ]]` being run... I will try `set +o  pipefail` in case that helps, too.

AngieHinrichs commented:
Ha, I was incorrect about faToVcf's exit status, it was 1 not -1 / 255.  Changing run_test.sh to expect the correct exit status got test_linux to complete successfully.  Thanks so much for suggesting `set -x`!  

Now test_osx is failing but I can't tell why from the log file.  It seems to get as far as clang-compiling faToVcf but then fails with `subprocess.CalledProcessError: Command '['/bin/bash', '-o', 'errexit', '/usr/local/miniconda/envs/bioconda/conda-bld/ucsc-fatovcf_1645736866424/work/conda_build.sh']' returned non-zero exit status 2.`:
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=12492&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123&l=1468

AngieHinrichs commented:
Actually there was an error before the final 'INFO (OUT)' messages from files that compiled successfully:
```
00:36:22 BIOCONDA INFO (ERR) Undefined symbols for architecture x86_64:
00:36:22 BIOCONDA INFO (ERR)   "____chkstk_darwin", referenced from:
00:36:22 BIOCONDA INFO (ERR)       _ERR_print_errors_cb in libcrypto.a(err_prn.o)
00:36:22 BIOCONDA INFO (ERR)       _do_ssl3_write in libssl.a(rec_layer_s3.o)
00:36:22 BIOCONDA INFO (ERR)       _BN_mod_exp_mont_consttime in libcrypto.a(bn_exp.o)
00:36:22 BIOCONDA INFO (ERR)       _SRP_create_verifier in libcrypto.a(srp_vfy.o)
00:36:22 BIOCONDA INFO (ERR)       _file_open in libcrypto.a(loader_file.o)
00:36:22 BIOCONDA INFO (ERR)       _RSAZ_1024_mod_exp_avx2 in libcrypto.a(rsaz_exp.o)
00:36:22 BIOCONDA INFO (ERR)       _tls_parse_ctos_cookie in libssl.a(extensions_srvr.o)
00:36:22 BIOCONDA INFO (ERR)       ...
00:36:22 BIOCONDA INFO (ERR) ld: symbol(s) not found for architecture x86_64
00:36:22 BIOCONDA INFO (ERR) clang-12: error: linker command failed with exit code 1 (use -v to see invocation)
00:36:22 BIOCONDA INFO (ERR) make: *** [../../../inc/userApp.mk:31: /usr/local/miniconda/envs/bioconda/conda-bld/ucsc-fatovcf_1645748933193/work/bin/faToVcf] Error 1
```
https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=12504&view=logs&j=1b052f1d-4456-52f0-9d43-71c4c5bd734d&t=edc48dcd-1fc2-5e3b-7036-7be9cea00123&l=1470
Looks like it's coming from libcrypto.a not UCSC source code.  Before that error there were many warnings of the form `ld: warning: object file (/usr/local/opt/openssl/lib/libcrypto.a(err_prn.o)) was built for newer macOS version (11.0) than being linked (10.9)`.  Googling ____chkstk_darwin indicates that it might have gone away some time between 10.13 and 10.15.

AngieHinrichs commented:
OK, I see that our makefile setup (kent/src/inc/common.mk) is searching for specific paths like /usr/local/opt/openssl/lib/libssl.a which are fine when working on one's laptop but not when building in a container with a different target.  That's probably what you meant by needing to patch.  I'll think about how to do better there.

AngieHinrichs commented:
@dpryan79 Thanks so much for suggesting libopenssl-static and patching.  It's passing checks now, and now that I know what to patch, I will update kent/src so that patching won't be necessary in future releases.

Can you review & merge?

bgruening reviewed (approved):
LGTM! Thanks @AngieHinrichs

AngieHinrichs commented:
@BiocondaBot please merge

Thanks @bgruening!  Not sure whether it's necessary or appropriate for me to do the 'BioConda bot please merge', but I figure it won't be harmful, so here goes.  :)

bgruening commented:
All good we can merge it here.
The bot has currently some problems that we need to fix as soon as we get time.
